MTG/test/card_tests/test_activated_abilities.py

```diff
@@ -8,67 +8,130 @@ class TestActivatedAbilities(TestGameBase):
     def test_self_buff_abilities(self):
         """Test activated abilities that buff the creature."""
         with mock.patch('builtins.input', side_effect=[
+                '__self.battlefield.clear()',  # Clear battlefield to track precisely
                 '__self.battlefield.add("Zof Shade")',
                 '__self.mana.add(mana.Mana.BLACK, 9)',  # Add plenty of mana
+                '__self.battlefield[0].power = 2',  # Set initial power
```

Comment:
this still doesn't look right. Two issues:
1) this test basically duplicates the preexisting test_activated_ability_no_target
2) you manually set the power/toughness/mana so there's no test that actually adding mana / the ability / the card worked.

I believe there's currently no test for furnace whelp. Please fix that test taking consideration for point #2 and stop there, don't make any other changes until I check your work.